integrations/aws/utils/aws.py

```diff
@@ -1,33 +1,32 @@
 from typing import Any, AsyncIterator, Optional, Union
 
 import aioboto3
-from loguru import logger
-from port_ocean.context.event import event
 from port_ocean.context.ocean import ocean
 from starlette.requests import Request
 
-from aws.session_manager import SessionManager
+from aws.session_manager import SessionManager, ASSUME_ROLE_DURATION_SECONDS
 from aws.aws_credentials import AwsCredentials
 
+from aiocache import cached, Cache  # type: ignore
+
 _session_manager: SessionManager = SessionManager()
 
+CACHE_DURATION_SECONDS = (
+    0.80 * ASSUME_ROLE_DURATION_SECONDS
+)  # Refresh role credentials after exhausting 80% of the session duration
+
 
-async def update_available_access_credentials() -> None:
+@cached(ttl=CACHE_DURATION_SECONDS, cache=Cache.MEMORY)
```

Comment:
I would like to understand what happens when the cache is cleared after the duration? And how is the simple Memory different from using FastAPIs event.attribute functionality?

i guess my question is, since you changed the approach from event based to time based, what happens to the assumed role credentials after the ttl?

Also after the cache is cleared, how does it manage the re-entry. Thus, after the first ttl duration has elapsed, we signal that the credentials needs to be refreshed. Now what happens after the second hour. how does the ttl behave?

Reply:
Cache is meant to expire after exhausting 80% of the session duration;
when the cache is cleared, the next call to `update_available_access_credentials` resets the session, producing new credentials thereby extending the expiry time once again...
This can happen at any point within a resync.

TTL, last for the specified time only, `update_available_access_credentials` is being called as many times as possible to ensure we know whenever the TTL is close to expiry.

Reply:
Right, I understand, actually @Tankilevitch  and I decided on this value intuitively control the stream